src/convexhull.jl

```diff
@@ -0,0 +1,30 @@
+"""
+Compute the upper part of the convex hull of a given set of 2D points 'pset' using the "Graham Scan" algorithm.
+
+*) Sort the points lexicographic
+
+*) Assess the last 3 points in 'upperHull'. If the the clockwise angle is <180° remove the second to last point.
+
+"""
+
+function upperhull(pset::PointSet) 
+
+  xs=sort(coordinates.(pset), by=first)
+
+  upperHull=[xs[1],xs[2]] 
+  
+  for i in 3:length(xs)
+    push!(upperHull,xs[i])
+    
+    # remove the second to last point in upperHull while the last 3 points make a "left turn"
+    while length(upperHull)>2 && ∠(Point(upperHull[lastindex(upperHull)-2]),Point(upperHull[lastindex(upperHull)-1]),Point(upperHull[lastindex(upperHull)]))<0 
```

Comment:
This line is very difficult to read. Notice how long it is. Please reorganize the code so that others can quickly understand what is going on. Create additional veriables if necessary outside the while loop. Also the syntax `upperHull[lastindex(upperHull)-2]` can be replaced by `upperHull[end-2]`

Reply:
I've shortened the stopping criterion, but had to define new variables inside the while loop, since they get updated inside the loop as well.